url0e7d57473a_kyorohiro_HetimaUtil_tgz-pJ8-net_hetimatan_net_http_HttpServer3xxJ8/src/net/hetimatan/util/http/HttpGetRequestUri.java

```diff
@@ -0,0 +1,344 @@
+
+		try {
+			MarkableReaderHelper.match(reader, "?".getBytes());
+			query(reader, ret);
+		} catch(IOException e) {
+		}
+		
+		try {
+			fragment(reader);
+		} catch(IOException e) {
+		}
+		
+		return ret;
+	}
+
+	public static String fragment(MarkableReader reader) throws IOException {
+		try {
+			MarkableReaderHelper.match(reader, "#".getBytes());
+			return "#" + new String(MarkableReaderHelper.asciiAndGet(reader, "".getBytes(), 256));
+		} catch(IOException e) {
+			throw e;
+		}
+	}
+
+	public static void query(MarkableReader reader, HttpGetRequestUri uri) throws IOException {
+		boolean first = true;
+		do {
+			try {
+				reader.pushMark();
+				if(!first) {
+					MarkableReaderHelper.match(reader, "&".getBytes());
+				} else {
+					first = false;
+				}
+				String key = new String(MarkableReaderHelper.asciiAndGet(reader, "=".getBytes(), Integer.MAX_VALUE));
+				MarkableReaderHelper.match(reader, "=".getBytes());
+				String value = new String(MarkableReaderHelper.asciiAndGet(reader, "& ".getBytes(), Integer.MAX_VALUE));
+				uri.putVale(key, value);
+			} catch(IOException e) {
+				reader.backToMark();
+				break;
+			} finally {
+				reader.popMark();				
+			}
+		} while(true);
+	}
+
+	public static String path(MarkableReader reader) throws IOException {
+		try {
+			MarkableReaderHelper.match(reader, "/".getBytes());
+			return "/"+new String(MarkableReaderHelper.asciiAndGet(reader, "? ".getBytes(), Integer.MAX_VALUE));//jumpAndGet(reader, available, 256));
+		} catch(IOException e) {
+			throw e;
+		}
+	}
+
+	public static int port(MarkableReader reader) throws IOException, NumberFormatException {
+		final byte[] available= {
+				'0', '1', '2', '3', '4', '5', '6', '7', '8', '9'
+		};
+
+		try {
+			reader.pushMark();
+			return Integer.parseInt(new String(MarkableReaderHelper.jumpAndGet(reader, available, 256)));
+		} catch(IOException e) {
+			reader.backToMark();
+			throw e;
+		} catch(NumberFormatException e) {
+			reader.backToMark();
+			throw e;
+		} finally  {
+			reader.popMark();
+		}
+	}
+
+	public static String host(MarkableReader reader) throws IOException {
+		final byte[] available= {
+				'0', '1', '2', '3', '4', '5', '6', '7', '8', '9',
+				'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j',
+				'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't',
+				'u', 'v', 'w', 'x', 'y', 'z',
+				'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J',
+				'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T',
+				'U', 'V', 'W', 'X', 'Y', 'Z',
+				'.', '-'
+		};
+
+		try {
+			return new String(MarkableReaderHelper.jumpAndGet(reader, available, 256));
+		} catch(IOException e) {
+			throw e;
+		}
+	}
+
+	public static String scheme(MarkableReader reader) throws IOException {
+		try {
+			MarkableReaderHelper.match(reader, ("https").getBytes());
+			return "https";
+		} catch(IOException e) {
+		}
+		try {
+			MarkableReaderHelper.match(reader, ("http").getBytes());
+			return "http";
+		} catch(IOException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)